### ISSUE TITLE
Do not clean dev cache on restart

### ISSUE THREAD

**Issue by ryardley** [blitz-js/blitz#226]
**What do you want and why do you want it?.**
Currently running `blitz start` will blow away the file cache next uses to run and rebuild it. This means that on huge projects when you stop blitz and then restart it the whole project needs to be copied to the cache which can take quite a bit of time. We should work out a way to prevent this and only copy changed files.

**Possible implementation(s)**
There are some ideas fleshed out in the [synchronizer readme](https://github.com/blitz-js/blitz/blob/canary/packages/server/src/synchronizer/README.md#future-thinking)

Basically we finish the optimizer to detect file system changes based on a single file we write. This file links the changes we have made to the filesystem back to the original file that spurred those changes. Whenever blitz starts it reads that file and uses it to determine if the file needs processing. This should probably be disabled on `blitz build`.
